pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -171,46 +163,121 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 							return
 						case <-time.After(controller.NamespaceResyncPeriod):
 							newRootCert := s.CA.GetCAKeyCertBundle().GetRootCertPem()
-							if !bytes.Equal(rootCert, newRootCert) {
-								rootCert = newRootCert
-								if err = ioutil.WriteFile(internalSelfSignedRootPath, rootCert, 0o600); err != nil {
-									log.Errorf("Failed to update local copy of self-signed root: %v", err)
+							if !bytes.Equal(caBundle, newRootCert) {
+								caBundle = newRootCert
```

Comment:
nit: can we break this out into a function? It feels very weird to modify the `caBundle` value in closure here. I think the code works today, but could easily lead to bugs.

Reply:
sure